src/core/meshcop/dtls.cpp

```diff
@@ -184,17 +184,11 @@ void Dtls::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageI
         ExitNow();
 
     case Dtls::kStateOpen:
-        IgnoreError(mSocket.Connect(Ip6::SockAddr(aMessageInfo.GetPeerAddr(), aMessageInfo.GetPeerPort())));
```

Comment:
`Connect()` causes Posix kernel to choose a fixed source address for the socket, which may not be the socket used by the peer. It would cause subsequent DTLS messages to be rejected (ICMPv6 Port unreachable).

Reply:
I read through the code..I think/agree we don't need the `Connect()` since we are going to be specify/provide all addresses/port in `messageInfo`.

Reply:
Just saw you note (after I added the last comment). Thanks. Make sense...:)

Reply:
We call `Connect()` to make sure the socket will only receive packets from the peer. This idea was borrowed from mbedTLS implementation.

I think it's OK to remove it for now before we figured out how to bind the correct source address.

Reply:
Agree with @bukepo 

I tried to close and reopen/rebind the socket to the correct infra link address and port, but it does not work because Udp6::Bind only works with a Thread netif address.

It will require significant efforts to make rebinding work, which don't have obvious benefit.